django_sharding_library/router.py

```diff
@@ -22,10 +22,16 @@ def get_specific_database_or_none(self, model):
         return getattr(model, 'django_sharding__database', None)
 
     def get_shard_for_instance(self, instance):
-        return instance._state.db or instance.get_shard()
+        try:
+            return instance._state.db or instance.get_shard()
+        except:
```

Comment:
ditto. I don't love the idea of a generic try/except unless there's a good reason for it. Did you hit anything specific?